src/config/firefox-desktop.js

```diff
@@ -112,11 +113,31 @@ export default {
         this.probeView[metricType] === 'categorical'
           ? 'proportion'
           : 'quantile';
-      const data = transformAPIResponse[viewType](
-        payload.response,
-        aggregationLevel,
-        metricType
-      );
+      let data = payload.response;
+
+      // Attach labels to histogram if appropriate type.
+      if (metricType === 'histogram-categorical') {
+        const labels = {
+          ...appStore.getState().probe.info.calculated.latest_history.details
+            .labels,
+        };
+        data = produce(data, (draft) => {
+          draft.forEach((point, i) => {
+            draft[i].histogram = Object.entries(point.histogram).reduce(
```

Comment:
I generally prefer to use `.map` instead of mutating things with`.forEach` where possible. This code may need some massaging for prettier, etc. but the general idea would be:

```suggestion
        data = produce(data, (draft) => {
          draft.map(point => {
            return {
                ...point,
                histogram: Object.entries(point.histogram).reduce(
```

Reply:
That looks great, thanks!